dot/parachain/availability-store/availability_store.go

```diff
@@ -420,89 +409,85 @@ func branchesFromChunks(chunks [][]byte) (branches, error) {
 }
 
 // Run runs the availability store subsystem
-func (av *AvailabilityStoreSubsystem) Run(ctx context.Context, OverseerToSubsystem chan any,
-	SubsystemToOverseer chan any) {
-
-	av.wg.Add(1)
-	go av.processMessages()
+func (av *AvailabilityStoreSubsystem) Run(ctx context.Context, overseerToSubsystem <-chan any) {
+	for {
+		select {
+		case msg := <-overseerToSubsystem:
+			logger.Infof("received message %T, %v", msg, msg)
+			av.processMessages(msg)
+		case <-time.After(av.pruningConfig.pruningInterval):
+			av.pruneAll()
+		case <-ctx.Done():
+			if err := ctx.Err(); err != nil {
+				logger.Errorf("ctx error: %v", err)
+			}
+			return
+		}
+	}
 }
 
 // Name returns the name of the availability store subsystem
 func (*AvailabilityStoreSubsystem) Name() parachaintypes.SubSystemName {
 	return parachaintypes.AvailabilityStore
 }
 
-func (av *AvailabilityStoreSubsystem) processMessages() {
-	for {
-		select {
-		case msg := <-av.OverseerToSubSystem:
-			logger.Infof("received message %T, %v", msg, msg)
-			switch msg := msg.(type) {
-			case QueryAvailableData:
-				err := av.handleQueryAvailableData(msg)
-				if err != nil {
-					logger.Errorf("failed to handle available data: %w", err)
-				}
-			case QueryDataAvailability:
-				err := av.handleQueryDataAvailability(msg)
-				if err != nil {
-					logger.Errorf("failed to handle query data availability: %w", err)
-				}
-			case QueryChunk:
-				err := av.handleQueryChunk(msg)
-				if err != nil {
-					logger.Errorf("failed to handle query chunk: %w", err)
-				}
-			case QueryChunkSize:
-				err := av.handleQueryChunkSize(msg)
-				if err != nil {
-					logger.Errorf("failed to handle query chunk size: %w", err)
-				}
-			case QueryAllChunks:
-				err := av.handleQueryAllChunks(msg)
-				if err != nil {
-					logger.Errorf("failed to handle query all chunks: %w", err)
-				}
-			case QueryChunkAvailability:
-				err := av.handleQueryChunkAvailability(msg)
-				if err != nil {
-					logger.Errorf("failed to handle query chunk availability: %w", err)
-				}
-			case StoreChunk:
-				err := av.handleStoreChunk(msg)
-				if err != nil {
-					logger.Errorf("failed to handle store chunk: %w", err)
-				}
-			case StoreAvailableData:
-				err := av.handleStoreAvailableData(msg)
-				if err != nil {
-					logger.Errorf("failed to handle store available data: %w", err)
-				}
-			case parachaintypes.ActiveLeavesUpdateSignal:
-				err := av.ProcessActiveLeavesUpdateSignal(msg)
-				if err != nil {
-					logger.Errorf("failed to process active leaves update signal: %w", err)
-				}
-			case parachaintypes.BlockFinalizedSignal:
-				err := av.ProcessBlockFinalizedSignal(msg)
-				if err != nil {
-					logger.Errorf("failed to process block finalized signal: %w", err)
-				}
+func (av *AvailabilityStoreSubsystem) processMessages(msg any) {
```

Comment:
this has now become processMessage :)
apply this in all subsystems when you have made this change

Reply:
updated